lib/PHPExif/Reader/Reader.php

```diff
@@ -50,11 +52,15 @@ public function __construct(AdapterInterface $adapter = null)
     /**
      * Setter for the reader adapter
      *
-     * @param \PHPExif\Reader\AdapterInterface $adapter
-     * @return \PHPExif\Reader Current instance for chaining
+     * @param \PHPExif\Adapter\AdapterInterface $adapter
+     * @return $this Current instance for chaining
+     * @throws ImmutableException when adapter is already set
      */
     public function setAdapter(AdapterInterface $adapter)
```

Comment:
If you want to make the ``Reader`` immutable, then it makes more sense to just remove the ``setAdapter`` altogether, and require an ``AdapterInterface`` in the constructor (remove ``null`` default).

For me, when a mutator (``setAdapter``) is available, it means that it can be used any time I want, which now isn't the case anymore.

It's perfectly ok for me to make the ``Reader`` class immutable; the ``setAdapter`` implicates otherwise, hence I would simply remove it.